alerts/project_triggers.go

```diff
@@ -146,21 +152,26 @@ func (trig TaskFailTransition) ShouldExecute(ctx triggerContext) (bool, error) {
 		q := alertrecord.ByLastFailureTransition(ctx.task.DisplayName, ctx.task.BuildVariant, ctx.task.Project)
 		lastAlerted, err := alertrecord.FindOne(q)
 
-		if err != nil {
+		if err != nil || lastAlerted == nil {
 			errMessage := getShouldExecuteError(ctx)
 			errMessage[message.FieldsMsgName] = "could not find a record for the last alert"
 			errMessage["error"] = err.Error()
+			errMessage["lastAlert"] = lastAlerted
 			grip.Error(errMessage)
 			return false, err
 		}
 
-		if lastAlerted == nil || lastAlerted.TaskId == "" {
+		if lastAlerted.TaskId == "" {
+			maybeSend := sometimes.Quarter()
 			errMessage := getShouldExecuteError(ctx)
-			errMessage[message.FieldsMsgName] = "last alert record nil, or empty last alert task_id"
+			errMessage[message.FieldsMsgName] = "empty last alert task_id"
 			errMessage["lastAlert"] = lastAlerted
-			errMessage["outcome"] = "sending alert"
-			grip.Info(errMessage)
-			return true, nil
+			if maybeSend {
+				errMessage["outcome"] = "sending alert 25% of the time"
+			}
+			grip.Warning(errMessage)
+
+			return maybeSend, nil
```

Comment:
Why the added return? Returning from the function only 25% of the time, only when we log, seems odd.

Reply:
we were always returning on old line 163. 

it's clearly not correct that we send alerts every time (e.g. returning true,) but without more logging it's hard to see why we're sending alerts each time. 

clarified message